docs/sources/next/javascript-api/k6-experimental/browser/page/dollar.md

```diff
@@ -44,7 +44,8 @@ export default async function () {
   const page = await browser.newPage();
 
   await page.goto('https://test.k6.io/browser.php');
-  await page.$('#text1').type('hello world');
+  const text = await page.$('#text1');
+  await text.type('hello world');
```

Comment:
It didn't apply `await` to two promises there when I tried. @allansson any thoughts?

Reply:
There's an issue with operator precedence here.

```ts
await page.$('#text').type('hello world')
```

This would try to call the function `type` on the returned promise. 

Putting the await in parenthesis resolves the precedence, but requires two awaits:

```ts
await (await page.$('#text')).type('hello world')
```

This would await the promise from `$`, call the `type` function on the resolved value and then await the promise from `type`. 

How I would write it:

```ts
await page
  .$('#text1')
  .then(text => text.type('hello world');
```

Reply:
As a separate thought: since the use of this is function discouraged, perhaps we shouldn't give an example of how to use it?

Reply:
I wouldn't say it's idiomatic. Some declare intermediates and put await in front of everything. I like `then` because it saves me the time of coming up with names of intermediates that don't collide with other relevant things.

Stick with what you have. You'll find people in both camps.

Reply:
From what I understand, this usage was fine:

```
const text = await page.$('#text1');
await text.type('hello world')
```